src/components/AddToCartModal/AddToCartModal.tsx

```diff
@@ -0,0 +1,76 @@
+import Modal from 'react-modal';
+import Image from 'next/image';
+import { Product } from '~/types/product';
+import Checkmark from '~/../public/authenticated/checkmark.svg';
+import Button from '~/components/Button';
+import closeModal from '~/../public/authenticated/close-modal.svg';
+
+export default function AddToCartModal({
+  openModal,
+  product,
+  onModalClose,
+}: {
+  openModal: boolean;
+  product: Product | null;
+  onModalClose: () => void;
+}) {
+  const modalStyles = {
+    content: {
+      margin: 'auto',
+      width: 'fit-content',
+      height: 'fit-content',
+      top: 32,
+      padding: 0,
+      display: 'inline-block',
+    },
+  };
+
+  return (
+    <Modal isOpen={openModal} style={modalStyles}>
+      <div className="flex h-121 flex-col items-center justify-between px-8 py-7 md:h-63 md:w-210 md:flex-row xl:mr-8 xl:w-265">
+        <div className="relative mr-8 h-44 w-65 rounded-lg border border-green-600 xl:mr-0">
+          <Image
+            src={product?.product_picture || ''}
+            alt=""
+            fill={true}
+            className="h-full w-full object-contain"
+          />
+          <Image
+            src={Checkmark}
+            alt=""
+            className="absolute bottom-0 right-0 h-12 w-12"
+          />
+        </div>
+        <div className="flex h-44 w-145 flex-col items-center justify-between text-dark-violet xl:mt-4">
+          <div className="text-2xl font-bold">
+            {`${product?.name} has been successfully added to your shopping cart!`}
```

Comment:
Can `product` be null? In what cases? Maybe you can control its existence

Reply:
Can you explain what you mean? I'd prefer if it couldn't be null, I used to have a guard for the whole modal to not load it if we had no product and in that case this wasn't needed, but later I removed it based on a comment from Agustín that it was a bad idea to have that guard

Reply:
Yes, I mean that if product could be null, then it will be displayed in the screen as `undefined`.  That maybe you can control it in the component, returning null if !product, or something

Reply:
@agustinmrgo, you tell me as I removed that because of your comment 😅